frontend/src/components/molecules/messages/components/Avatar.tsx

```diff
@@ -28,7 +38,9 @@ const MessageAvatar = ({ author, hide }: Props) => {
     if (isAssistant && selectedChatProfile?.icon) {
       return selectedChatProfile.icon;
     }
-    return apiClient?.buildEndpoint(`/avatars/${author || 'default'}`);
+
+    const authorAvatarId = getAuthorAvatarId(author);
```

Comment:
i don't think doing this in the client is a good idea since developers also use chainlit with custom frontends

Reply:
Maybe we can give custom frontend implementors (2-line!) instructions to implement this?

This patch would not only improve security, but being consistent in how we deal with filenames will make the developer experience much more robust. This would replace a `.` (and other characters, think unicode) with `_`, allowing _any_ characters in avatar names.

We currently do not: a lot of author names will prevent the avatar from working. This patch solves that, fully, without compromising security again.

Reply:
Proposed compromise:

1. We keep allowing spaces and upper characters (basically not change the backend).
2. We throw a deprecation error every time an 'invalid' URL comes in.
3. We update the frontend code already.

This would let existing users know of what's coming, recover pre-existing functionality in cases of special characters, guarantee sufficient security for now and maximal security long-term.